### DISCUSSION TITLE
python（Nested calls） pack to exe ,not same as expected，sub module not pack success.

### DISCUSSION THREAD

**Discussion by liujian163** [pyinstaller/pyinstaller#8914] [Help] [answered]
python main script would parse json data inluding pythn script name and parameters to excute sub python script, use pyinstaller  latest version V6.11.1 to pack main script，had try to usee --onefile or dir to generate，use Analysis include sub script，hiddenimport python module used in sub script，but all not valid to report not find module like xlrd, cantools when excute sub script ，just want to pack python and main python script and sub script to one exe for use，pyinstall now not friendly to me， please help，Dear pyinstaller gods，Thanks.

**Comment by liujian163** [comment]
hiddenimports=['xlrd', 'cantools', 'pandas'],

**Comment by bwoodsend** [comment, marked as answer]
Using a subprocess is always the wrong thing to do. Just import it or use runpy.

**Reply by liujian163** [reply]
yes, it's subporess poblem, new cmd can not use exe environment。
Thanks。

**Reply by rokm** [reply]
You can only use the embedded python environment that exe uses from exe itself. 

So if you absolutely must use a subprocess, then you need to re-design your program to accept arguments that switch it into "script running mode" (for example, `--run-script <script-path>`) instead of running your main code. This script running mode then loads the script and executes it. In your main program code, you should spawn the sub-process with `[sys.executable, '--run-script`, path_to_script]` as the arguments. See https://github.com/orgs/pyinstaller/discussions/5978#discussioncomment-971765 for a proof-of-concept example.

But do note that such external sub-scripts do not undergo import analysis, so you will need to manually specify all its imports as `hiddenimports`, otherwise they might not be collected (i.e., if they are not imported elsewhere, for example in the main application script) and will end up missing.
